test/models/user_test.rb

```diff
@@ -63,4 +63,18 @@ class UserTest < ActiveSupport::TestCase
       assert_equal n.archived, true
     end
   end
+
+  test "triggers sync_notifications on save" do
+    notifications_url = %r{https://api.github.com/notifications}
+
+    body     = '[]'
+    headers  = { 'Content-Type' => 'application/json' }
+    response = { status: 200, body: body, headers: headers }
```

Comment:
This is starting to become a pattern, we may want to look into a helper for this.